src/java/fr/paris/lutece/plugins/forms/web/admin/FormTransitionJspBean.java

```diff
@@ -565,15 +567,41 @@ private boolean validateTransition( Transition transition )
     private ReferenceList getTransitionTargetStepReferenceList( int nIdForm, int nIdStep )
     {
         ReferenceList listTransitionTargetSteps = new ReferenceList( );
-
+        List<Transition> transitions = TransitionHome.getTransitionsListFromForm( nIdForm );
         for ( ReferenceItem step : StepHome.getStepReferenceListByForm( nIdForm ) )
         {
-            if ( NumberUtils.toInt( step.getCode( ) ) != nIdStep )
+
+                if ( ( transitions.isEmpty( ) && NumberUtils.toInt( step.getCode( ) ) != nIdStep )
+                        || validateStep ( NumberUtils.toInt( step.getCode( ) ), nIdStep, transitions  ) )
             {
                 listTransitionTargetSteps.add( step );
             }
         }
         return listTransitionTargetSteps;
     }
 
+    /**
+     * @param nStepToValidate
```

Comment:
a description of the method should be added to say that it validates the steps to be put in the form for creating a transition